internal/resources/azure/api_management.go

```diff
@@ -0,0 +1,122 @@
+package azure
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+
+	"fmt"
+	"strings"
+
+	"github.com/shopspring/decimal"
+)
+
+type APIManagement struct {
+	Address                string
+	Region                 string
+	SKUName                string
+	SelfHostedGatewayCount *int64 `infracost_usage:"self_hosted_gateway_count"`
+	MonthlyAPICalls        *int64 `infracost_usage:"monthly_api_calls"`
+}
+
+var APIManagementUsageSchema = []*schema.UsageItem{{Key: "self_hosted_gateway_count", ValueType: schema.Int64, DefaultValue: 0}, {Key: "monthly_api_calls", ValueType: schema.Int64, DefaultValue: 0}}
+
+func (r *APIManagement) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *APIManagement) BuildResource() *schema.Resource {
+	region := r.Region
+
+	costComponents := []*schema.CostComponent{}
+	skuName := r.SKUName
+	var tier string
+	var capacity decimal.Decimal
+	if s := strings.Split(skuName, "_"); len(s) == 2 {
+		tier = strings.ToLower(s[0])
+		capacity, _ = decimal.NewFromString(s[1])
+	}
+
+	if tier != "consumption" {
+		costComponents = append(costComponents, apiManagementCostComponent(
+			fmt.Sprintf("API management (%s)", tier),
+			"units",
+			region,
+			tier,
+			&capacity))
+
+	} else {
+		var apiCalls *decimal.Decimal
+		if r.MonthlyAPICalls != nil {
+			apiCalls = decimalPtr(decimal.NewFromInt(*r.MonthlyAPICalls))
+		}
+
+		if apiCalls != nil {
+			apiCalls = decimalPtr(apiCalls.Div(decimal.NewFromInt(10000)))
+			costComponents = append(costComponents, consumptionAPICostComponent(region, tier, apiCalls))
+		} else {
+			costComponents = append(costComponents, consumptionAPICostComponent(region, tier, apiCalls))
+		}
+	}
+
+	if tier == "premium" {
+		var selfHostedGateways *decimal.Decimal
+		if r.SelfHostedGatewayCount != nil {
+			selfHostedGateways = decimalPtr(decimal.NewFromInt(*r.SelfHostedGatewayCount))
+		}
+		costComponents = append(costComponents, apiManagementCostComponent(
+			"Self hosted gateway",
+			"gateways",
+			region,
+			"Gateway",
+			selfHostedGateways,
+		))
+	}
+
+	return &schema.Resource{
+		Name:           r.Address,
+		CostComponents: costComponents, UsageSchema: APIManagementUsageSchema,
+	}
+}
+
+func apiManagementCostComponent(name, unit, region, tier string, quantity *decimal.Decimal) *schema.CostComponent {
```

Comment:
Can the functions in this file be on the struct and therefore not need all the args?

Reply:
Applied